net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi/avahi.inc

```diff
@@ -130,3 +150,44 @@ function avahi_deinstall_command() {
 
 	unlink_if_exists(AVAHI_RCFILE);
 }
+
+function avahi_get_carp_list() {
+	$list = array();
+	$list['none'] = 'none';
+	$carplist = get_configured_vip_list("all", VIP_CARP);
+	foreach ($carplist as $vid => $vaddr) {
+		$vip = get_configured_vip($vid);
+		$list[$vid] = "$vaddr (vhid: {$vip['vhid']})";
+	}
+	return ($list);
+}
+
+function avahi_plugin_carp($pluginparams) {
+	global $config;
+	/* If avahi is not configured, then stop */
+	if (is_array($config['installedpackages']['avahi']['config'])) {
+		$avahi_config = &$config['installedpackages']['avahi']['config'][0];
+	} else {
+		return null;
+	}
+	/* If there is no properly configured CARP status check IP, then stop */
+	if (!isset($avahi_config['carpstatusvid']) || $avahi_config['carpstatusvid'] == "none") {
+		return null;
+	}
+	/* Locate the VIP and match it against the configured check address */
+	list($vhid, $iface) = explode("@", trim($pluginparams['interface']));
+	$friendly = convert_real_interface_to_friendly_interface_name($iface);
+	$vip = get_configured_vip($avahi_config['carpstatusvid']);
+	if ($vip['vhid'] != $vhid || $vip['interface'] != $friendly) {
```

Comment:
The individual tests should be enclosed in parenthesis:
```
(($vip['vhid'] != $vhid) || ($vip['interface'] != $friendly))
```

Reply:
done